test/unit/test_grouped_loader.py

```diff
@@ -0,0 +1,238 @@
+"""
+Tests for deepreg/dataset/loader/grouped_loader.py in
+pytest style
+"""
+from os.path import join
+
+import numpy as np
+import pytest
+
+from deepreg.dataset.loader.grouped_loader import GroupedDataLoader
+from deepreg.dataset.loader.h5_loader import H5FileLoader
+from deepreg.dataset.loader.nifti_loader import NiftiFileLoader
+
+FileLoaderDict = dict(nifti=NiftiFileLoader, h5=H5FileLoader)
+nifti_path = join("data", "test", "nifti", "grouped")
+h5_path = join("data", "test", "h5", "grouped")
+DataPaths = dict(nifti=nifti_path, h5=h5_path)
+
+
+def test_validate_data_files():
+    """
+    Test validate_data_files function looks for inconsistencies in the fixed/moving image and label lists.
+    If there is any issue it will raise an error, otherwise it returns None.
+    """
+    for key_file_loader, file_loader in FileLoaderDict.items():
+        for train_split in ["train", "test"]:
+            for labeled in ["True", "False", 1, 0]:
```

Comment:
`labeled` should be a bool, maybe we should change `if self.labeled:` to `if self.labeled is True:` in grouped_loader.py

Reply:
agreed, changed

Reply:
btw @agrimwood need to push? lol